### ISSUE TITLE
Improve tag feature #3 : creating a tag database

### ISSUE THREAD

**Issue by Globulle** [diaspora/diaspora#6078]
Following issues #6076 and #6077, I propose a third step to avoid tag synonyms and gain efficiency. It is probably the most complicated one, but I think it would be really helpful. We could create a referencing database [as for example in TeX StackExchange](http://meta.stackexchange.com/tags). For each tag it could display 3 fields:
* the number of occurrence within the *n* last days, to help (new) people to choose some of the most popular ones (see #5060).
* a brief description which could be edited by everybody, keeping track of the history and needing validation by *n* users. It would help using tags efficiently, especially knowing what they refer to when they are not so obvious. We could proceed as follow: when using a tag which is not referenced yet, user could be prompted to optionally add a description when hovering it (as well as being able to follow it as proposed in #4170).
* the list of synonyms: as for description, one could "declare this tag as duplicate of..." when hovering it. This would be particularly useful for names: we could use indifferently #Tolkien, #JRRTolkien, #JRR_Tolkien, or #JohnRonaldReuelTolkien .
Finally, hovering a tag could allow the 'follow' action (issue #4170) for a common tag, and would give access to 3 actions: 'follow', 'add a description' or 'declare as duplicate' when the tag is orphan.

Regarding federation, my first guess is that a copy of such a database could exist on each pod, with the advantage of making easier the tag suggestions introduced earlier in #6076. Copy would be synchronized and editing conflicts could be resolved by creating duplicate to be examined later. Hope this is not too complicated, I'm not clearly aware of technical problems.

**Comment by jhass** [comment]
I think that would be too complicated for the social network and the problem would already be solved by your other two suggestions and #6012

**Comment by Flaburgan** [comment]
Yeah as discussed IRL I think your two first propositions would already improve the situation a lot and its probably not worth the effort to implement this one: it would need a good UX, it asks for users interaction... So probably too complicated.

**Comment by Globulle** [comment]
OK, I agree. Let's see how things work when the two first will be implemented.
